gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py

```diff
@@ -0,0 +1,183 @@
+    mt = mt.filter_rows(mt.AF > 0.01)
+
+    logger.info(
+        "Filtering to variants with at least one het nonref call and checkpointing..."
+    )
+    mt = mt.filter_rows(hl.agg.sum(mt.het_non_ref) >= 1)
+
+    logger.info(
+        "Adding new genotype entries with original homalt hotfix and het nonref-aware hotfix..."
+    )
+    mt = mt.annotate_entries(
+        GT_hotfix=hl.if_else(
+            mt.GT.is_het() & (mt.AF > 0.01) & (mt.AD[1] / mt.DP > 0.9),
+            hl.call(1, 1),
+            mt.GT,
+        ),
+        GT_hetnonref_fix=hl.if_else(
+            mt.GT.is_het()
+            & ~mt.het_non_ref
+            & (mt.AF > 0.01)
+            & (mt.AD[1] / mt.DP > 0.9),
+            hl.call(1, 1),
+            mt.GT,
+        ),
+    )
+    logger.info("Checking homozygote counts using the different genotypes...")
+    # Taking index 1 of gt_stats because it contains an entry for the reference allele at index 0
+    # homozygote_count (tarray of tint32) -
+    # Homozygote genotype counts for each allele, including the reference. Only diploid genotype calls are counted.
+    # (from hail docs, https://hail.is/docs/0.2/aggregators.html#hail.expr.aggregators.call_stats)
+
+    mt = mt.annotate_rows(gt_stats=hl.agg.call_stats(mt.GT_hotfix, mt.alleles))
+    mt = mt.transmute_rows(hom_count_hotfix=mt.gt_stats.homozygote_count[1])
+
+    mt = mt.annotate_rows(gt_stats=hl.agg.call_stats(mt.GT_hetnonref_fix, mt.alleles))
+    mt = mt.transmute_rows(hom_count_hetnonref_fix=mt.gt_stats.homozygote_count[1])
+
+    logger.info(
+        "Filtering to rows where homalt hotfix erroneously adjusts het nonref calls..."
+    )
+    mt = mt.filter_rows(mt.hom_count_hotfix != mt.hom_count_hetnonref_fix)
+
+    return mt.rows()
+
+
+def main(args):
+    """
+    Script used to get variants impacted by homalt hotfix.
+    Most of the work was done in notebooks, and any results created in notebooks are noted with comments.
+    """
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
+    ).select_entries(*SPARSE_ENTRIES)
+
+    if args.test:
+        logger.info("Filtering to two partitions on chr20")
+        mt = hl.filter_intervals(mt, [hl.parse_locus_interval("chr20:1-1000000")])
+        mt = mt._filter_partitions(range(2))
+
+    logger.info(
+        "Adding annotation for whether original genotype (pre-splitting multiallelics) is het nonref..."
+    )
+    # Adding a Boolean for whether a sample had a heterozygous non-reference genotype
+    # Need to add this prior to splitting MT to make sure these genotypes
+    # are not adjusted by the homalt hotfix downstream
+    mt = mt.annotate_entries(het_non_ref=mt.LGT.is_het_non_ref())
+    # TODO: Need to fix main code to add this annotation and use it also, where to add?
```

Comment:
I'm not as familiar with the gnomad pipeline, but for ukbb, it would've made sense to add this annotation to the hardcalls MT

Reply:
Yeah we don't have a hardcalls MT, so I think it will actually need to be added to the frequency code, but that will be another PR and I just wanted to make sure my thought was written down in the patch somewhere